astropy/time/tests/test_precision.py

```diff
@@ -1,28 +1,176 @@
+from hypothesis.extra.numpy import array_shapes, arrays
+from hypothesis.strategies import (composite, datetimes, floats, integers,
+                                   one_of, sampled_from, timedeltas)
+
 import numpy as np
-from decimal import Decimal
-from datetime import datetime
 
-import astropy.units as u
 import astropy._erfa as erfa
-from astropy.time import Time, TimeDelta
+import astropy.units as u
+from astropy.tests.helper import assert_quantity_allclose
+from astropy.time import STANDARD_TIME_SCALES, Time, TimeDelta
 from astropy.time.utils import day_frac, two_sum
 from astropy.utils import iers
-from astropy.utils.exceptions import ErfaWarning
+from astropy.utils.exceptions import ErfaError, ErfaWarning
 
-allclose_jd = functools.partial(np.allclose, rtol=2. ** -52, atol=0)
-allclose_jd2 = functools.partial(np.allclose, rtol=2. ** -52,
-                                 atol=2. ** -52)  # 20 ps atol
-allclose_sec = functools.partial(np.allclose, rtol=2. ** -52,
-                                 atol=2. ** -52 * 24 * 3600)  # 20 ps atol
+allclose_jd = functools.partial(np.allclose, rtol=np.finfo(float).eps, atol=0)
+allclose_jd2 = functools.partial(np.allclose, rtol=np.finfo(float).eps,
+                                 atol=np.finfo(float).eps)  # 20 ps atol
+allclose_sec = functools.partial(np.allclose, rtol=np.finfo(float).eps,
+                                 atol=np.finfo(float).eps * 24 * 3600)
 
-tiny = 2. ** -52
+tiny = np.finfo(float).eps
 dt_tiny = TimeDelta(tiny, format='jd')
 
 
+@pytest.fixture
+def iers_b():
+    with iers.earth_orientation_table.set(iers.IERS_B.open(iers.IERS_B_FILE)):
+        yield
+
+
+@contextlib.contextmanager
+def quiet_erfa():
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore", category=ErfaWarning)
+        yield
+
+
+def assert_almost_equal(a, b, *, rtol=None, atol=None, label=''):
+    """Assert numbers are almost equal
+
+    This version does this with two separate assertions, one that triggers with
+    severe failures (by a factor of a hundred), and another that triggers with
+    normal failures. The goal is that hypothesis should recognize that these
+    are different situations - severe failure versus marginal failure - and
+    avoid shrinking severe failures down to marginal ones.
+
+    This version also lets hypothesis know how far apart the inputs are, so
+    that it can work towards a failure (and, ultimately, perhaps present the
+    worst failure ever seen as well as the simplest, which often just barely
+    exceeds the threshold).
+    """
+    __tracebackhide__ = True
+    if rtol is None or rtol == 0:
+        thresh = atol
+    elif atol is None:
+        thresh = rtol*(abs(a)+abs(b))/2
+    else:
+        thresh = atol + rtol*(abs(a)+abs(b))/2
+
+    amb = (a-b)
+    if isinstance(amb, TimeDelta):
+        ambv = amb.to_value(u.s)
+        target(ambv, label=label + " amb")
+        target(-ambv, label=label + " bma")
```

Comment:
Because these measures are exactly inverse, Hypothesis can maintain a *much* smaller Pareto front if you only specify one of them, i.e.:

```python
if ambv >= 0:
	target(ambv, label=label + " amb")
else:
    target(-ambv, label=label + " bma")
```